### PR TITLE
Fix `import-profile` constraints 

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes
## Purpose
This PR addresses a critical issue with the `import-profile` constraints. Currently, the IDs `import-profile-has-href-attribute` and `import-profile-has-valid-content` do not accurately represent the functionality of the constraints, nor do they align with the [constraint tracker](https://docs.google.com/spreadsheets/d/1m_ttH3VCRRCcd3rPc3CwkmNI3T74HW2cnWszIGJ8dzo/edit?pli=1&gid=0#gid=0). Also, the current implementation of the `import-profile-has-href-attribute` is incorrect. This constraint should use the `resolve-uri()` method and should also be able to handle a reference to a profile in the `back-matter`.

## Fixes
- Changed ID `import-profile-has-href-attribute` to `import-profile-has-available-document`.
- Changed ID `import-profile-has-valid-content` to `import-profile-resolves-to-fedramp-content`.
- Improved `expect` test for `import-profile-has-available-document` constraint to effectively resolve a uri and handle a reference to a profile in the back-matter.
- Changed all occurrences of old IDs to new IDs.
- Added more content files for `import-profile-has-available-document-PASS.yaml` to ensure it handles all possible cases.

## Testing
- Verified that all existing unit tests related to `import-profile` constraints still pass.
- Verified that new content files for `import-profile-has-available-document-PASS.yaml` pass.

{Please provide a description of what this PR accomplishes. Be sure to reference any issues addressed. If the PR is a work-in-progress submitted for early review, please submit the PR as a draft PR using the "Draft pull request" dropdown.}

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
~~- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?~~ This is a bug fix and does not require any changes to documentation. 
- [x] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Comment by wandmagic** [comment]
I think these import profile issues have to be handled upstream, much like implimented requirements

**Review by aj-stein-gsa — COMMENTED** [review]
Apparently I left these comments in the chamber and forgot about them.

**Comment by aj-stein-gsa** [comment]
> I think these import profile issues have to be handled upstream, much like implimented requirements

We are talking about a release with https://github.com/metaschema-framework/metaschema-java/issues/208 integrated, correct?

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -200,11 +200,11 @@
...
                 <message>A FedRAMP SSP MUST import a profile or catalog with a valid file or HTTP(S) address.</message>
             </expect>
-            <expect id="import-profile-has-valid-content" target="." test="count(resolve-profile(doc(resolve-uri($resolved-import-profile-href)))//control//prop[@ns='https://fedramp.gov/ns/oscal' and @name='response-point']) >= 2" level="CRITICAL">
+            <expect id="import-profile-resolves-to-fedramp-content" target="." test="count(resolve-profile(doc(resolve-uri($resolved-import-profile-href)))//control//prop[@ns='https://fedramp.gov/ns/oscal' and @name='response-point']) >= 2" level="CRITICAL">
                 <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="https://automate.fedramp.gov/documentation/ssp/3-working-with-oscal-files/#importing-the-fedramp-baseline"/>
                 <message>A FedRAMP SSP MUST import a profile or catalog of security controls to reference implemented requirements against those control(s).</message>
```
[the remark this answers is not in the document]
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml:209]: That is valid. Let me know if something like this is what you had in mind: [02b3fc8](https://github.com/GSA/fedramp-automation/pull/854/commits/02b3fc89f821ccde17cb28a728ae747e273ec5e9)

**Review by Rene2mt — APPROVED** [review]

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (aj-stein-gsa) [reply src/validations/constraints/fedramp-external-constraints.xml:209]: I will comment below just re style guide concerns this looks good beyond missing BCP14 language.

File: src/validations/constraints/unit-tests/import-profile-resolves-to-fedramp-content-PASS.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+test-case:
+  name: Positive Test for import-profile-resolves-to-fedramp-content
+  description: >-
+    This test case validates the behavior of constraint import-profile-resolves-to-fedramp-content.
+    Scenario 1 tests: a URI reference to a resolved profile catalog.
+    Scenario 2 tests: a URI reference to a profile.
+    Scenario 3 tests: a fragment that points to a resolved profile catalog in the back-matter.
+    Scenario 4 tests: a fragment that points to a profile in the back-matter.
```
Comment (aj-stein-gsa) [inline src/validations/constraints/unit-tests/import-profile-resolves-to-fedramp-content-PASS.yaml:8]: Non-blocking: excellent, this is the kind of test description I like to see!

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 3]
```diff
@@ -200,13 +200,16 @@
...
+                <remarks>
+                    <p>FedRAMP requires the use of a valid FedRAMP catalog when referencing security controls. Using catalogs from non-FedRAMP sources or external catalogs is not permitted.</p>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:211]: A good start, but one last tweak, based on FRR111 and FRR112 guidance.

```suggestion
                    <p>A FedRAMP SSP MUST use a valid FedRAMP catalog to reference security controls. It SSP MUST NOT reference controls from a non-FedRAMP catalog.</p>
```

**Review by aj-stein-gsa — APPROVED** [review]
I would like a minor tweak around some missing things in the message, but otherwise good.

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 3 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: Resolved! [b4cbe2b](https://github.com/GSA/fedramp-automation/pull/854/commits/b4cbe2be95421755f1d5bfefca54a16a46a63aa6) 🎊

**Review by aj-stein-gsa — APPROVED** [review]

**Review by Rene2mt — APPROVED** [review]

**Review by wandmagic — APPROVED** [review]

**Pull request merged**